introduction_and_learning/Histogram/histograms.py

```diff
@@ -0,0 +1,291 @@
+__all__ = [
+    "HistogramTxt",
+    "HistogramJSON",
+    "HistogramXml"
+]
+
+from histogram_api import HistogramInterface
+import json
+import xml.etree.ElementTree as ET
+import csv
+
+
+class HistogramTxt(HistogramInterface):
+    def read_data(self, file_path: str) -> str:
+        extension = file_path[file_path.rfind("."):]
+
+        if extension == ".txt":
+            with open(file_path, "tr", encoding="utf8") as f:
+                return f.read()
+        else:
+            return "File does not have a .txt extension."
+
+    def create(self, data_input: str) -> dict:
+        # Note: Delete from string characters "." and "," then transform in to
+        # ordered list.
+        words = data_input.casefold().replace(".", "").replace(",", "").split(" ")
+        words.sort()
+        # --- end note
+        hist = {}
+
+        for word in words:
+            if word in hist:
+                hist[word] += 1
+            else:
+                hist[word] = 1
+
+        return hist
+
+    def generate_report(self, histogram: dict, output_type: str = "html", path: str = "report") -> bool:
+        output_types = ["html", "csv", "txt"]
+
+        if output_type not in output_types:
+            return False
+        else:
+            if output_type == "html":
+                output = ["<html>\n<body>\n\t<table>\n\t<tr><th>Name</th>"
+                          "<th>Amount</th></tr>"]
+
+                for key in histogram.keys():
```

Comment:
To get keys and values at the same time you can use:
`for key, value in histogram.items():`